nix/digital-ocean.nix

```diff
@@ -0,0 +1,47 @@
+{ config, pkgs, lib, utils, ... }:
+
+with utils;
+with lib;
+with import ./lib.nix lib;
+
+let
+  cfg = config.deployment.digital-ocean;
+in
+{
+  ###### interface
+  options = {
+
+    deployment.digital-ocean.authToken = mkOption {
+      default = "";
+      example = "8b2f4e96af3997853bfd4cd8998958eab871d9614e35d63fab45a5ddf981c4da";
+      type = types.str;
+      description = ''
+        The API auth token. We're checking the environment for
+        <envar>DIGITAL_OCEAN_API_TOKEN</envar> first and if that is
+        not set we try this auth token.
+      '';
+    };
+
+    deployment.digital-ocean.region = mkOption {
+      default = "";
+      example = "nyc3";
+      type = types.str;
+      description = ''
+        The region. See https://status.digitalocean.com/ for a list
+        of regions.
+      '';
+    };
+
+    deployment.digital-ocean.size = mkOption {
+      example = "512mb";
+      type = types.str;
```

Comment:
It might be nicer to use an integer here (e.g. size in megabytes) rather than an opaque string.